edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/persistence/entity/TransportCall.java

```diff
@@ -0,0 +1,47 @@
+package org.dcsa.edocumentation.domain.persistence.entity;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import lombok.*;
+import org.dcsa.skernel.domain.persistence.entity.base.BaseTransportCall;
+import org.hibernate.annotations.Formula;
+
+import javax.persistence.*;
+import java.time.OffsetDateTime;
+
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Setter(AccessLevel.PRIVATE)
+@Entity
+public class TransportCall extends BaseTransportCall {
+  @ToString.Exclude
+  @EqualsAndHashCode.Exclude
+  @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinColumn(name = "vessel_id")
+  private Vessel vessel;
+
+  @ToString.Exclude
+  @EqualsAndHashCode.Exclude
+  @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinColumn(name = "import_voyage_id")
+  private Voyage importVoyage;
+
+  @ToString.Exclude
+  @EqualsAndHashCode.Exclude
+  @ManyToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinColumn(name = "export_voyage_id")
+  private Voyage exportVoyage;
+
+  @Formula("(select transport_event.event_date_time from transport_event where transport_event.transport_call_id = id AND" +
```

Comment:
WTF. You need to get this from the event??? Isn't event is supposed to be created based on this, not the other way around.

Reply:
The events get created based on the schedules. A schedule leads to a lot of PLN (Planned) events to be created - it is these events that are used in order to create the TransportPlan